component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRest.java

```diff
@@ -513,11 +514,12 @@ public Response getComments(
                                                                            sortDescending,
                                                                            offset,
                                                                            limit);
-    CollectionEntity collectionComment = new CollectionEntity(commentsEntity, EntityBuilder.COMMENTS_TYPE, offset, limit);
+    ActivityCommentCollectionEntity collectionComment = new ActivityCommentCollectionEntity(commentsEntity, EntityBuilder.COMMENTS_TYPE, offset, limit);
     if (returnSize) {
       boolean expandSubComments = EntityBuilder.expandSubComments(expand);
       RealtimeListAccess<ExoSocialActivity> listAccess = activityManager.getCommentsWithListAccess(activity, expandSubComments);
       collectionComment.setSize(listAccess.getSize());
+      collectionComment.setTotalCommentsSize(activityManager.getNumberOfAllComments(activityId));
```

Comment:
You may need to add this with a dedicated condition `if (expandFields.contains(TOTAL_COMMENTS_COUNT)) {}` as made for activity.

Reply:
done